Specialization/Havoc.lua

```diff
@@ -1,486 +1,230 @@
-	local covenantId = fd.covenant.covenantId;
-
-	MaxDps:GlowCooldown(HV.Metamorphosis, cooldown[HV.Metamorphosis].ready);
-
-	if talents[HV.Momentum] then
-		MaxDps:GlowCooldown(HV.FelRush, not buff[HV.MomentumAura].up and cooldown[HV.FelRush].ready);
-		--MaxDps:GlowCooldown(HV.VengefulRetreat, not buff[HV.MomentumAura].up and cooldown[HV.VengefulRetreat].ready);
-	end
-
-	-- metamorphosis,if=!(talent.demonic.enabled|variable.pooling_for_meta)&cooldown.eye_beam.remains>20&(!covenant.venthyr.enabled|!dot.sinful_brand.ticking)|fight_remains<25;
-	--if cooldown[HV.Metamorphosis].ready and (not (talents[HV.Demonic] or poolingForMeta) and cooldown[HV.EyeBeam].remains > 20 and (not covenantId == Enum.CovenantType.Venthyr or not debuff[HV.SinfulBrand].up) or fightRemains < 25) then
-	--	return HV.Metamorphosis;
-	--end
-
-	-- metamorphosis,if=talent.demonic.enabled&(cooldown.eye_beam.remains>20&(!variable.blade_dance|cooldown.blade_dance.remains>gcd.max))&(!covenant.venthyr.enabled|!dot.sinful_brand.ticking);
-	--if cooldown[HV.Metamorphosis].ready and (talents[HV.Demonic] and (cooldown[HV.EyeBeam].remains > 20 and (not bladeDance or cooldown[HV.BladeDance].remains > gcd)) and (not covenantId == Enum.CovenantType.Venthyr or not debuff[HV.SinfulBrand].up)) then
-	--	return HV.Metamorphosis;
-	--end
-
-	-- sinful_brand,if=!dot.sinful_brand.ticking;
-	if covenantId == Venthyr and cooldown[HV.SinfulBrand].ready and (not debuff[HV.SinfulBrand].up) then
-		return HV.SinfulBrand;
-	end
-
-	-- the_hunt,if=!talent.demonic.enabled&!variable.waiting_for_momentum|buff.furious_gaze.up;
-	if covenantId == NightFae and
-		cooldown[HV.TheHunt].ready and
-		currentSpell ~= HV.TheHunt and
-		(not talents[HV.Demonic] and not waitingForMomentum or buff[HV.FuriousGaze].up)
-	then
-		return HV.TheHunt;
-	end
+local function IsUnstoppableConflictUp()
+	local fd = MaxDps.FrameData
+	local up = MaxDps:UnitAura(HV.UnstoppableConflictDebuff, fd.timeShift, "player", "HARMFUL")
+	return up ~= nil and up
+end
 
-	-- fodder_to_the_flame;
-	if covenantId == Necrolord and cooldown[HV.FodderToTheFlame].ready then
-		return HV.FodderToTheFlame;
-	end
+function DemonHunter:Havoc()
+	local fd = MaxDps.FrameData
+	local cooldown = fd.cooldown
+	local talents = fd.talents
+	local buff = fd.buff
+	local fury = GetFury()
+	local furyMax = UnitPowerMax('player', Enum.PowerType.Fury)
+	local furyDeficit = furyMax - fury
+	fd.fury = fury
+	fd.furyMax = furyMax
+	fd.furyDeficit = furyDeficit
+
+	local saveGlaiveForFodderToTheFlameDemon = buff[HV.FodderToTheFlameBuff].up and IsUnstoppableConflictUp()
+	fd.saveGlaiveForFodderToTheFlameDemon = saveGlaiveForFodderToTheFlameDemon
+
+	local furyGenerator
+	if talents[HV.Felblade] and cooldown[HV.Felblade].ready then
+		furyGenerator = HV.Felblade
+	elseif not talents[HV.DemonBlades] then
+		furyGenerator = HV.DemonsBite
+	end
+	fd.furyGenerator = furyGenerator
+
+	local poolingForMeta = cooldown[HV.Metamorphosis].remains < 6 and furyDeficit > 30
```

Comment:
I probably have it in mine too as a hold over from previous coding but why have a local you never use? 

Wouldn't this make more sense? 
fd.poolingForMeta = cooldown[HV.Metamorphosis].remains < 6 and furyDeficit > 30